client/question/question.js

```diff
@@ -822,3 +822,19 @@ tie.constant('CODE_RESET_CONFIRMATION_MESSAGE', [
   'Are you sure you want to reset your code? This will clear all of your ',
   'existing progress on the question.'
 ].join(''));
+
+/**
+ * Constant that denotes the compile-time execution context.
+ *
+ * @type {string}
+ * @constant
+ */
+tie.constant('EXECUTION_CONTEXT_COMPILATION', 'COMPILATION');
```

Comment:
Something about "Compilation" versus "Run with Tests" bothers me a smidge. Do you think it should be Compile and Run With Tests, so that they're both verbs?

Reply:
I don't think execution contexts should be verbs. I can't find a clearer way to describe them though (than what I currently have). Do you have any specific suggestions?

If not, I suggest keeping them as-is, as long as it's clear to the reader.

Reply:
yeah that's fine I suppose.